django/core/handlers/base.py

```diff
@@ -137,20 +201,83 @@ def _get_response(self, request):
 
         return response
 
-    def process_exception_by_middleware(self, exception, request):
+    async def _get_response_async(self, request):
         """
-        Pass the exception to the exception middleware. If no middleware
-        return a response for this exception, raise it.
+        Resolve and call the view, then apply view, exception, and
+        template_response middleware. This method is everything that happens
+        inside the request/response middleware.
         """
-        for middleware_method in self._exception_middleware:
-            response = middleware_method(request, exception)
+        response = None
+        callback, callback_args, callback_kwargs = self.resolve_request(request)
+
+        # Apply view middleware.
+        for middleware_method in self._view_middleware:
+            response = await middleware_method(request, callback, callback_args, callback_kwargs)
             if response:
-                return response
-        raise
+                break
+
+        if response is None:
+            wrapped_callback = self.make_view_atomic(callback)
+            # If it is a synchronous view, run it in a subthread
+            if not asyncio.iscoroutinefunction(wrapped_callback):
+                wrapped_callback = sync_to_async(wrapped_callback, thread_sensitive=True)
+            try:
+                response = await wrapped_callback(request, *callback_args, **callback_kwargs)
+            except Exception as e:
+                response = await sync_to_async(self.process_exception_by_middleware, thread_sensitive=True)(e, request)
```

Comment:
I'm not sure if wrapping with `sync_to_async()` is required :thinking: , because it's already adapted in [BaseHandler.load_middleware()](https://github.com/django/django/pull/11650/files#diff-dbd7d9159676b15fc9a096b0adb919e9R82-R84). I don't see any failure after changing to:
```python
response = await self.process_exception_by_middleware(e, request)
```

Reply:
You can't just `await` a non-async function like that, unfortunately, it won't work. The exception methods were adapted, but the wrapper was not; I've instead changed it for now so `process_exception` is always run synchronously, to avoid duplicating this request path (and since it does not need to be super performance-sensitive anyway).